.github/workflows/ci-zap.yml

```diff
@@ -15,11 +15,11 @@ jobs:
       name: dev
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```

Comment:
I wonder if these are required to use a specific node version.
```
- uses: actions/setup-node@v3
  with:
    node-version: 16
```

Reply:
Sure, I will test this out.

Reply:
It did not work, produced an limit error and still showed the warning about zaproxy/action-full-scan.  To see the warning just look at last failed under ZAP in actions. I think this may be a action-full-scan issue. I found this yesterday, it is recent with the person having the same warning issue.  https://github.com/zaproxy/action-full-scan/issues/61

You can see that they implemented the node upgrade internally. So I don't know if this is fixable at this point.

Reply:
According to the following announcement, I think we don't need to specify github action's local node version if it's 16.
[All Actions will begin running on Node16 instead of Node12](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/).

I can't find the warning from the action log, so wonder where it comes.

btw, I don't know why but the action takes too long and fails sometimes.

Reply:
Yeah, that's why just adding the latest @v makes it use Node 16.  The maintainers of the different action repos internally set them to use Node 16. That is why there are no warnings on any other action, only the zaproxy/action-full-scan@v0.4.0. Which I mentioned, didn't update node 16 internally this version @0.4.0. 

I don't know what could be done about it taking too long other than increasing the timeout. This is the zaproxy warning, the others would not have warnings.

https://github.com/bcgov/internationally-educated-nurses/actions/runs/4408883243

Reply:
Hmm, maybe instead of updsting the action versions, we just set `setup-node` to 16 in each workflow, and go back to v2.  There are some new changes in v3 actions that could be hampering performance

Reply:
Let's use `setup-node` only when the action uses nodejs. I found that `setup-node to 16` actually downgrades the version of ubuntu image.